lib/feeder_importer.rb

```diff
@@ -71,25 +160,138 @@ def create_series(pcast = podcast)
-      url: feeder_podcast_url,
+      url: podcast_id,
       distributable: series,
       audio_version_template: template
     )
 
     series
   end
 
-  def retrieve_podcast(a_id = account_id, p_url = feeder_podcast_url)
-    self.podcast = api(root: feeder_root, account: a_id).tap { |a| a.href = p_url }.get
+  def create_series_image(podcast_image, purpose)
+    upload_url = copy_image(SecureRandom.uuid, podcast_image)
+    image = series.images.create!(upload: upload_url, purpose: purpose)
+    announce_image(image)
+    podcast_image.update_attribute(:original_url, image.public_url(version: 'original'))
+  end
+
+  def create_stories
+    podcast.episodes.each do |episode|
+      create_story(episode)
+    end
+  end
+
+  def create_story(episode)
+    # puts "episode.categories: #{episode.categories.inspect}"
+    attrs = {
+      app_version: PRX::APP_VERSION,
+      creator_id: user_id,
+      account_id: account_id,
+      title: episode.title,
+      short_description: episode.subtitle,
+      description_html: episode.description,
+      tags: episode.categories,
+      published_at: episode.published_at
+    }
+    story = series.stories.create!(attrs)
+
+    version = story.audio_versions.create!(
+      audio_version_template: template,
+      explicit: episode.explicit
+    )
+
+    episode.media_resources.each_with_index do |media_resource, i|
+      upload_url = copy_media(episode, media_resource)
+      audio = version.audio_files.create!(label: "Segment #{i + 1}", upload: upload_url)
+      announce_audio(audio)
+      media_resource.update_attribute(:original_url, audio_file_original_url(audio))
+    end
+
+    episode.episode_images.each do |episode_image|
+      upload_url = copy_image(episode.guid, episode_image)
+      image = story.images.create!(upload: upload_url)
+      announce_image(image)
+      episode_image.update_attribute(:original_url, image.public_url(version: 'original'))
+    end
+
+    # create the story distribution
+    episode_url = "#{feeder_root}/episodes/#{episode.guid}"
+    story.distributions << StoryDistributions::EpisodeDistribution.create!(
+      distribution: distribution,
+      story: story,
+      guid: episode.item_guid,
+      url: episode_url
+    )
+
+    if set_episode_urls
+      new_url = default_url(story)
+      episode.update_attribute(:url, new_url)
+    end
+    episode.update_attribute(:prx_uri, "/api/v1/stories/#{story.id}")
+    stories << story
+
+    story
+  end
+
+  def default_url(story)
+    path = "#{story.class.name.underscore.pluralize}/#{story.id}"
+    ENV['PRX_HOST'].nil? ? nil : "https://#{ENV['PRX_HOST']}/#{path}"
+  end
+
+  def audio_file_original_url(af)
+    "s3://#{Rails.env}.mediajoint.prx.org/public/audio_files/#{af.id}/#{File.basename(af.filename)}"
+  end
+
+  def update_podcast
+    podcast.update_attributes(
+      prx_account_uri: "/api/v1/accounts/#{account_id}",
+      prx_uri: "/api/v1/series/#{series.id}",
```

Comment:
if I'm understanding your comment at top of file correctly, setting the podcast's prx_uri to the link with the newly created series id in it will get us to the point where changing things in CMS will change them in Feeder too, right? In which case, should this be a separate method that we call manually after `import` has completed?

Reply:
That is the effect it will have, but I did not see a compelling reason to wait and add an additional manual step.

The point of import is to get to that point, and I designed this to all work in a single script run rather than needing to come back and fix more things later.

We can be extra cautious and do this manually later, but I didn't want to do that and have any changes in cms discarded by feeder until we set it.